File: multikube-rest-service/src/main/java/com/multikube_rest_service/dtos/responses/provider/ClusterResponse.java
package com.multikube_rest_service.dtos.responses.provider;

import com.multikube_rest_service.common.enums.ClusterStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ClusterResponse {
    private Long id;
    private String name;
    private String description;
    private Long providerUserId;
    private String providerUsername;
    private ClusterStatus status;
    private Timestamp createdAt;
    private Timestamp updatedAt;
}

File: multikube-ui/src/app/common/rest/types/provider/responses/ClusterResponse.ts
export interface ClusterResponse {
    id: number,
    name: string,
    description: string,
    providerUserId: number,
    providerUsername: string,
    status: ClusterStatus,
    createdAt: string,
    updatedAt: string
    allocation?: ClusterAllocation;
}

export interface ClusterAllocation {
    tenantName: string;
    tenantId: number
}

export enum ClusterStatus {
    PENDING_VERIFICATION = "PENDING_VERIFICATION", // Initial status upon registration
    ACTIVE = "ACTIVE",               // Verified and reachable
    UNREACHABLE = "UNREACHABLE",          // Verification failed or lost connectivity
    DEGRADED = "DEGRADED",             // Partially reachable or issues detected
    INACTIVE = "INACTIVE",             // Manually set to inactive by provider
    DELETING = "DELETING",             // Marked for deletion
    ERROR = "ERROR"                 // An unspecified error state
}
